goal_src/jak2/engine/debug/nav/mysql-nav-graph.gc

```diff
@@ -5,58 +5,1154 @@
+   (node-array              mysql-nav-node-array                  :offset-assert   8)
+   (edge-array              mysql-nav-edge-array                  :offset-assert  12)
+   (visnode-array           mysql-nav-visnode-array               :offset-assert  16)
+   (level-info-array-length int32                                 :offset-assert  20)
+   (level-info-last-lookup  int32                                 :offset-assert  24)
+   (level-info-array        mysql-nav-graph-level-info 32 :inline :offset-assert  28)
+   )
+  :method-count-assert 21
+  :size-assert         #x41c
+  :flag-assert         #x150000041c
+  (:methods
+    (new (symbol type int string) _type_ 0)
+    (init-from-sql! (_type_ string string) symbol 9)
+    (exec-sql! (_type_) symbol 10)
+    (indexof-nav-node (_type_ int) int 11)
+    (indexof-nav-edge (_type_ int) int 12)
+    (alloc-new-node! (_type_) int 13)
+    (alloc-new-edge! (_type_) int 14)
+    (indexof-visnode (_type_ int int) int 15)
+    (alloc-new-visnode! (_type_ int int) int 16)
+    (mysql-nav-graph-method-17 (_type_) none 17)
+    (lookup-level-info2 (_type_ mysql-nav-node symbol) mysql-nav-graph-level-info 18)
+    (mysql-nav-graph-method-19 (_type_) none 19)
+    (mysql-nav-graph-method-20 (_type_) none 20)
+    )
+  )
+
+
+;; WARN: Return type mismatch object vs mysql-nav-graph.
+(defmethod new mysql-nav-graph ((allocation symbol) (type-to-make type) (arg0 int) (arg1 string))
+  (let ((gp-0 (object-new allocation type-to-make (the-as int (-> type-to-make size)))))
+    (if (zero? gp-0)
+        (return (the-as mysql-nav-graph 0))
+        )
+    (set! (-> gp-0 node-array) (new 'debug 'mysql-nav-node-array 3000))
+    (set! (-> gp-0 node-array length) 0)
+    (set! (-> gp-0 edge-array) (new 'debug 'mysql-nav-edge-array 5000))
+    (set! (-> gp-0 edge-array length) 0)
+    (set! (-> gp-0 visnode-array) (new 'debug 'mysql-nav-visnode-array 3000))
+    (set! (-> gp-0 visnode-array length) 0)
+    (let ((s4-0 (clear *temp-string*)))
+      (format *temp-string* "select nav_graph_id from nav_graph where name='~S'" arg0)
+      (let ((a2-6 (sql-query s4-0)))
+        (when (!= (-> a2-6 error) 'select)
+          (format 0 "ERROR: sql: select error ~A for ~A~%" a2-6 gp-0)
+          (return (the-as mysql-nav-graph #f))
+          )
+        (set! (-> gp-0 nav_graph_id) (the-as uint (string->int (the-as string (-> a2-6 data 0)))))
+        )
+      )
+    gp-0
+    )
+  )
+
+(defmethod alloc-new-node! mysql-nav-graph ((obj mysql-nav-graph))
+  "Allocates a new `[[mysql-nav-node]]`, if `node-array`'s `length` exceeds `3000` return `-1`
+       otherwise, return the new size of the array"
+  (cond
+    ((>= (-> obj node-array length) 3000)
+     (format #t "mysql-nav-graph : nodes buffer too small, increase NAV_GRAPH_EDITOR_NODE_COUNT~%")
+     -1
+     )
+    (else
+      (let ((gp-0 (the-as mysql-nav-node (+ (+ (* 80 (-> obj node-array length)) 12) (the-as int (-> obj node-array)))))
```

Comment:
hopefully this indexing is fixed after the node-array type has the nodes `inline`.

Reply:
I wasn't able to fix `(method 10 mysql-nav-graph)` still, but the casts are kinda tricky so it might be my fault -- changing everything to `:inline` did fix the other method though...which is why im leaning on blaming myself, but im not quite sure where i made a mistake.